src/endpoint/endpoint.js

```diff
@@ -337,16 +374,15 @@ function setup_http_server(server) {
     // });
 }
 
-function update_virtual_host_suffix(base_address) {
-    if (process.env.CONTAINER_PLATFORM === 'KUBERNETES') {
-        dbg.warn(`skip update of virtual host suffix in kuberentes. support will be added in the future`);
-        return;
-    }
-    const suffix = base_address && url.parse(base_address).hostname;
-    const new_virtual_suffix = net_utils.is_fqdn(suffix) ? '.' + suffix : undefined;
-    if (new_virtual_suffix !== virtual_host_suffix) {
-        virtual_host_suffix = new_virtual_suffix;
-        dbg.log0('The virtual host suffix of this agent was changed to this:', virtual_host_suffix, 'base_address', base_address);
+function update_virtual_hosts(new_virtual_hosts) {
+    new_virtual_hosts = new_virtual_hosts
+        .filter(suffix => net_utils.is_fqdn(suffix))
+        .sort();
```

Comment:
Removed the sort from the user code (system_store.on('load', () => ...) left it here. 
The filter still needed because the list can come for a diff source (not only the system_store list)